File: SNS24.API/Models/Appointments/Appointment.cs
using SNS24.WebApi.Models;

namespace SNS24.Api.Models.Appointments;

public class Appointment : BaseEntity
{
    public DateTime Date { get; set; }
    public bool Attended { get; set; }

    // fk
    public Guid? DoctorId { get; set; }
    public Doctor? Doctor { get; set; }

    public Guid PatientId { get; set; }
    public Patient Patient { get; set; }

    public Guid InstitutionId { get; set; }
    public Institution? Institution { get; set; }
}

File: webapp/src/app/shared/interfaces/medical-appointment/appointment.ts
import { Doctor } from "../doctor/doctor";
import { Institution } from "../institution";
import { Patient } from "../patient/patient";

export interface Appointment {
  id: string;
  date: string; // ISO string
  attended: boolean;
  doctorId: string;
  doctor: Doctor;
  patientId: string;
  patient: Patient;
  institutionId: string;
  institution?: Institution;
}
